src/main/java/com/icthh/xm/uaa/security/oauth2/idp/jwk/JwkDefinitionSource.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.icthh.xm.uaa.security.oauth2.idp.jwk;
+
+import org.springframework.security.jwt.codec.Codecs;
+import org.springframework.security.jwt.crypto.sign.EllipticCurveVerifier;
+import org.springframework.security.jwt.crypto.sign.RsaVerifier;
+import org.springframework.security.jwt.crypto.sign.SignatureVerifier;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.KeyFactory;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.RSAPublicKeySpec;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class copied from {@link org.springframework.security.oauth2.provider.token.store.jwk.JwkDefinitionSource}.
+ * Reason: we need to provide an ability of custom implementation of JwkDefinitionSource class
+ * which is impossible to import and override cause it has package private access.
+ *
+ * A source for JSON Web Key(s) (JWK) that is solely responsible for fetching (and caching)
+ * the JWK Set (a set of JWKs) from the URL supplied to the constructor.
+ *
+ * @see JwkSetConverter
+ * @see JwkDefinition
+ * @see SignatureVerifier
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7517#page-10">JWK Set Format</a>
+ *
+ * @author Joe Grandja
+ * @author Michael Duergner
+ */
+public class JwkDefinitionSource {
```

Comment:
looks like library class, maybe better to add library to gradle that copy paste source code?

Reply:
It won't solve problems with package private access in the class definition. 
Also, some methods, ex. JwkDefinitionSource#getDefinitionLoadIfNecessary should have public access to provide the ability to override their behaiviour.
Also, JwkDefinitionSource should have a default constructor for correct work of custom XmJwkDefinitionSource. which is missing in the original class.